lib/stack_master/template_compiler.rb

```diff
@@ -1,8 +1,11 @@
 module StackMaster
   class TemplateCompiler
+    TemplateCompilationFailed = Class.new(RuntimeError)
 
     def self.compile(config, template_file_path)
       template_compiler_for_file(template_file_path, config).compile(template_file_path)
+    rescue StandardError
```

Comment:
What happens when an error is wrapped in Ruby is, the original error is available in the wrapped errors `#cause`... so we could check and use that (if it's available) in the error message we display to the user.